File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/BookApi.java
package ru.elenacherev.librarymanager.api;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import ru.elenacherev.librarymanager.api.dto.Book;
import ru.elenacherev.librarymanager.api.dto.BookUsing;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public interface BookApi {
    Book getBook(
            @NotNull Long bookId
    );

    Page<Book> searchBooks(
            Pageable pageable
    );

    Book createBook(
            @Valid Book book
    );

    Book saveBook(
            @NotNull Long bookId,
            @Valid Book book
    );
    
    Page<BookUsing> searchBookUsings(
    		@NotNull Long bookId,
            Pageable pageable
    );
    
    List<BookUsing> updateBookUsings(
    		@NotNull Long bookId,
    		@NotNull List<Long> bookUsingIds
    );
}

File: librarymanager-api/src/main/kotlin/ru/elenacherev/librarymanager/api/BookApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Book
import ru.elenacherev.librarymanager.api.dto.BookUsing
import javax.validation.Valid

interface BookApi {
    fun getBook(
        bookId: Long
    ): Book?

    fun searchBooks(
        pageable: Pageable
    ): Page<Book>

    fun createBook(
        book: @Valid Book
    ): Book

    fun saveBook(
        bookId: Long,
        book: @Valid Book
    ): Book

    fun searchBookUsings(
        bookId: Long,
        pageable: Pageable
    ): Page<BookUsing>

    fun updateBookUsings(
        bookId: Long,
        bookUsingIds: List<Long>
    ): List<BookUsing>
}